object/search.go

```diff
@@ -60,33 +143,29 @@ func SearchMatchFromV2(t v2object.MatchType) SearchMatchType {
 //   - [MatchUnknown]: MATCH_TYPE_UNSPECIFIED.
 //
 // All other values are base-10 integers.
-func (m SearchMatchType) EncodeToString() string {
-	return v2object.MatchType(m).String()
-}
+// Deprecated: use [SearchMatchTypeToString] instead.
```

Comment:
Why a new function for this?

Reply:
we agreed to do this for similar types before

Reply:
In fact what most people would really appreciate here is good old `fmt.Stringer`.

Reply:
these are helpers providing auxiliary mapping around a type that is not inherently textual, so it goes beyond the interface in essence, and in the proposed version also in the code

Stringer iface is for printing and doesnt undertake to maintain compatibility

Reply:
The problem is, no one cares. There is https://pkg.go.dev/encoding#TextMarshaler for really pedantic ones, but even that is rarely used/implemented. While https://pkg.go.dev/fmt#Stringer can be found just about everywhere and there is `go:generate stringer` that is also used a lot. Compatibility is what _you_ define, not fmt.Stringer or encoding.TextMarshaler, `String()` can be compatible and can be not, different matter.

I can merge this as it is now and deal with it later (since we already have some code like this), but I think we're seriously raising WTF/minute metric for SDK users with this. Instead of relying on simple interfaces everyone knows about and expects we're creating some local rules that are hardly justified and that developers need to grasp before they can use the SDK.

Reply:
unfortunately or fortunately there is no inverse interface for the stringer, so it needs to be invented in any case and it will not be familiar to the user anyway. Helper functions pair best emphasize the essence of this mapping. Godoc always puts such functions in the section close to the type so they are easy to detect. And the stringer simply produces printed readable text as described in its docs